workflow/executor/pns/pns.go

```diff
@@ -202,68 +197,54 @@ func (p *PNSExecutor) Wait(ctx context.Context, containerID string) error {
 // It opens file handles on all root pids because at this point, we do not yet know which pid is the
 // "main" container.
 // Polling is necessary because it is not possible to use something like fsnotify against procfs.
-func (p *PNSExecutor) pollRootProcesses(timeout time.Duration) {
-	log.Warnf("Polling root processes (%v)", timeout)
-	deadline := time.Now().Add(timeout)
+func (p *PNSExecutor) pollRootProcesses(ctx context.Context, containerNames []string) {
```

Comment:
The previous implementation limited our hard tight-looped polling to 1 minute, because it was quite aggressive. How do we limit this in the new implementation?

Reply:
Good point. I’ve removed this by accident and need to add it back.

Reply:
this is fixed now, I find it is easier to do timeouts using `context.WaitTimeout` because you usually have a context available